### PR TITLE
feat: add referral

### LINKED ISSUE #1: add referral
https://github.com/puncsky/web-beancount/issues/107

### PR DESCRIPTION
**Pull request by yuyongmao**
**What type of PR is this?**

> Uncomment only one `/kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind ui-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**Which issue(s) this PR fixes**:

Fixes #1

### CODE REVIEW

File: src/screens/referral-screen/invite-screen.tsx [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+        </View>
+      );
+    }
+    return (
+      <View style={styles().flex1}>
+        <SectionList
+          showsVerticalScrollIndicator={false}
+          bounces={false}
+          sections={sections}
+          renderSectionHeader={({ section }) => (
```
Comment (puncsky) [inline src/screens/referral-screen/invite-screen.tsx:172]: having issues with dark mode

File: src/screens/referral-screen/components/contact-row.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  onPress,
+  name,
+  emailOrNumber,
+  selected,
+}: ContactRowProps): JSX.Element {
+  return (
+    <TouchableOpacity activeOpacity={0.9} onPress={onPress}>
+      <View style={styles().rowContainer}>
+        <Text style={{ marginRight: contentPadding }}>
+          {selected ? "✅" : "⭕️"}
```
Comment (puncsky) [inline src/screens/referral-screen/components/contact-row.tsx:35]: can we use https://mobile.ant.design/components/checkbox/ ?

Comment (yuyongmao) [reply src/screens/referral-screen/components/contact-row.tsx]: change  checked and unchecked icon

File: src/screens/referral-screen/invite-screen.tsx (hunk 1 above)
Comment (yuyongmao) [reply src/screens/referral-screen/invite-screen.tsx:226]: already support dark mode

**Review by puncsky — APPROVED** [review]

**Pull request merged**